File: bindings/lua/Polycode/BinaryObjectWriter.lua
class "BinaryObjectWriter"


function BinaryObjectWriter:addKey(key)
	local retVal = Polycode.BinaryObjectWriter_addKey(self.__ptr, key)
	return retVal
end

function BinaryObjectWriter:getKeyIndex(key)
	local retVal = Polycode.BinaryObjectWriter_getKeyIndex(self.__ptr, key)
	return retVal
end

function BinaryObjectWriter:writeToFile(fileName)
	local retVal = Polycode.BinaryObjectWriter_writeToFile(self.__ptr, fileName)
	return retVal
end

function BinaryObjectWriter:__delete()
	if self then Polycode.delete_BinaryObjectWriter(self.__ptr) end
end

File: bindings/javascript/Polycode/BinaryObjectWriter.js
function BinaryObjectWriter() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.BinaryObjectWriter()
	}
}


Duktape.fin(BinaryObjectWriter.prototype, function (x) {
	if (x === BinaryObjectWriter.prototype) {
		return;
	}
	Polycode.BinaryObjectWriter__delete(x.__ptr)
})

BinaryObjectWriter.prototype.addKey = function(key) {
	return Polycode.BinaryObjectWriter_addKey(this.__ptr, key)
}

BinaryObjectWriter.prototype.getKeyIndex = function(key) {
	return Polycode.BinaryObjectWriter_getKeyIndex(this.__ptr, key)
}

BinaryObjectWriter.prototype.writeToFile = function(fileName) {
	return Polycode.BinaryObjectWriter_writeToFile(this.__ptr, fileName)
}
